File: uk-ets-app-registry/uk-ets-registry-web/src/app/shared/user/keycloak-user.ts
export interface KeycloakUser {
  username: string;
  email: string;
  firstName: string;
  lastName: string;
  eligibleForSpecificActions: boolean;
  userRoles: string[];
  attributes: {
    urid: string[];
    alsoKnownAs: string[];
    buildingAndStreet: string[];
    buildingAndStreetOptional: string[];
    buildingAndStreetOptional2: string[];
    country: string[];
    countryOfBirth: string[];
    postCode: string[];
    townOrCity: string[];
    stateOrProvince: string[];
    birthDate: string[];
    state: string[];
    workBuildingAndStreet: string[];
    workBuildingAndStreetOptional: string[];
    workBuildingAndStreetOptional2: string[];
    workCountry: string[];
    workCountryCode: string[];
    workEmailAddress: string[];
    workEmailAddressConfirmation: string[];
    workPhoneNumber: string[];
    workPostCode: string[];
    workTownOrCity: string[];
    workStateOrProvince: string[];
    lastLoginDate?: string[];
    memorablePhrase?: string;
  };
}

File: uk-ets-app-reports-generator/src/main/java/gov/uk/ets/reports/generator/domain/KeycloakUser.java
package gov.uk.ets.reports.generator.domain;

import java.time.LocalDateTime;
import java.util.List;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class KeycloakUser {

    private String urid;
    private String keycloakUserId;
    private List<UserRole> role;

    private LocalDateTime createdOn;
    private LocalDateTime registeredOn;
    private LocalDateTime lastLoginOn;
    private String username;
    private String email;
    private String status;
    private String firstName;
    private String lastName;
    private String alsoKnownAs;
    private String birthDate;
    private String country;
    private String countryOfBirth;
    private String buildingAndStreet;
    private String buildingAndStreetOptional;
    private String buildingAndStreetOptional2;
    private String postCode;
    private String townOrCity;
    private String stateOrProvince;
    private String workCountry;
    private String workCountryCode;
    private String workPhoneNumber;
    private String workBuildingAndStreet;
    private String workBuildingAndStreetOptional;
    private String workBuildingAndStreetOptional2;
    private String workPostCode;
    private String workTownOrCity;
    private String workStateOrProvince;

}
